### PR TITLE
Integrate new renewal date API in cancellation form

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Integrate new API for renewal date in cancellation form.

Update date input to handle rendering without a date.

Update chevron icon to match figma.

## Justify why they are needed

We should use the renewal date for cases where the user is selecting the expiry date of their current insurance.

We need to make some API updates so "bank signering" options are based on these dates rather than start date.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1545** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1545" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1545?utm_source=stack-comment).

File: apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ import { CancellationForm, CancellationOption } from './CancellationForm/Cancell
 import * as ComparisonTable from './ComparisonTable/ComparisonTable'
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx:23]: Nit: This module keeps growing, is it time to split?

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ProductPage/PurchaseForm/OfferPresenter.tsx:23]: Yes indeed -- just wanna let it grow until it stabilizes and then refactor

**Pull request merged**